chain/src/txhashset/txhashset.rs

```diff
@@ -1456,8 +1457,31 @@ pub fn zip_write(
 ) -> Result<(), Error> {
 	let txhashset_path = Path::new(&root_dir).join(TXHASHSET_SUBDIR);
 	fs::create_dir_all(txhashset_path.clone())?;
-	zip::decompress(txhashset_data, &txhashset_path, expected_file)
-		.map_err(|ze| ErrorKind::Other(ze.to_string()))?;
+
+	// catch the panic to avoid the thread quit
+	{
+		panic::set_hook(Box::new(|_info| {
+			// do nothing
+		}));
+		let result = panic::catch_unwind(|| {
```

Comment:
`zip::decompress` is our code (in the `util` crate).
Would it make sense to push this `catch` wrapper logic down into the util crate and make this a more general purpose solution?

I'm not sure I really like having this low-level `catch_unwind` in our `txhashset` code.

Reply:
Accept.
Move catch to the `util` crate.